x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/audit/logfile/LoggingAuditTrailFilterTests.java

```diff
@@ -1851,6 +1958,88 @@ private ClusterSettings mockClusterSettings() {
+            "indices:data/write/delete*",
+            "indices:data/write/*",
+            "indices:monitor/*",
+            "indices:admin/*",
+            "indices:admin/ilm/*",
+            "indices:admin/refresh*",
+            "indices:admin/flush*",
+            "indices:admin/synced_flush",
+            "indices:admin/forcemerge*",
+            "indices:admin/index_template/*",
+            "indices:admin/data_stream/*",
+            "indices:admin/template/*",};
+        Random random = random();
+        for (int i = 0; i < randomIntBetween(1, 4); i++) {
+            Object name = actionPatterns[random.nextInt(actionPatterns.length)];
+            filtered.add((String)name);
+        }
+        return filtered;
+    }
+
+    private List<String> randomNonEmptyListOfFilteredClusterActions() {
+        final List<String> filtered = new ArrayList<>(4);
+        final String[] actionPatterns = {
+            "cluster:admin/xpack/security/*",
+            "cluster:admin/xpack/security/saml/*",
+            "cluster:admin/xpack/security/oidc/*",
+            "cluster:admin/xpack/security/token/*",
+            "cluster:admin/xpack/security/api_key/*",
+            "cluster:monitor/*",
+            "cluster:monitor/xpack/ml/*",
+            "cluster:monitor/text_structure/*",
+            "cluster:monitor/data_frame/*",
+            "cluster:monitor/xpack/watcher/*",
+            "cluster:monitor/xpack/rollup/*",
+            "cluster:*",
+            "cluster:admin/xpack/ml/*",
+            "cluster:admin/data_frame/*",
+            "cluster:monitor/data_frame/*",
+            "cluster:monitor/transform/*",
+            "cluster:admin/transform/*",
+            "cluster:admin/xpack/watcher/*",
+            "cluster:monitor/nodes/liveness",
+            "cluster:monitor/state",
+            "cluster:admin/component_template/*",
+            "cluster:admin/ingest/pipeline/*",
+            "cluster:admin/xpack/rollup/*",
+            "cluster:admin/xpack/ccr/*",
+            "cluster:admin/ilm/*",
+            "cluster:admin/slm/*",
+            "cluster:admin/xpack/enrich/*"};
+        Random random = random();
+        for (int i = 0; i < randomIntBetween(1, 4); i++) {
+            Object name = actionPatterns[random.nextInt(actionPatterns.length)];
+            filtered.add((String)name);
+        }
+        return filtered;
+    }
+
+    private List<String> randomNonEmptyListOfFilteredPrivileges(List<String> listOfActions) {
+        final List<String> filtered = new ArrayList<>();
+        for (int i = 0; i < listOfActions.size(); i++) {
+            Collection<String> privileges = AuthorizationService.isIndexAction(listOfActions.get(i)) ?
+                IndexPrivilege.findPrivilegesThatGrant(listOfActions.get(i)) :
+                ClusterPrivilegeResolver.findPrivilegesThatGrant(listOfActions.get(i));
+            assertNotNull(privileges);
+            filtered.addAll(privileges);
+        }
+        return filtered;
+    }
+
```

Comment:
I don't think the unit tests are adequate for the change we're making here.
They feel like they're mostly just adding some "action" based filters to existing tests, but this change has some significant additional requirements that need to be tested. 

Some tests that I would expect to see:
1. A filter of `cluster_privileges: all` discards all cluster events
2. A filter of `index_privileges: create_doc` works correctly (because create doc is special)
3. A filter of `cluster_privileges: manage_own_api_key` works correctly (because it is special)
4. More functional testing such as a filter of `index_privileges: index` ignores ingestion but logs deletes.

Reply:
Added more tests to cover all cornel cases + more functional